luarules/gadgets/unit_seismic_dots_and_advanced_radar.lua

```diff
@@ -0,0 +1,363 @@
+	-- So, SetUnitPosErrorParams is on a *Per Unit* basis, and *every* team in the game sees the same radar dot position.
+	-- [Technically, the same radar dot Angle, and the distance from true unit center is scaled based on the number of pinpointers on the team]
+	-- Which means, we cannot show "low wobble" to team A and "high wobble" to team C at the same time.
+	-- Therefore, until deeper engine changes are made, the "radar wobble" of a unit will be based on the "best" radar of all enemy teams that is painting the unit
+	-- So, a unit on team B in advanced radar via Team A, but in seismic in Team C, will exhibit low wobble to *both* teams
+	-- Should only be seen in FFA and other formats with more than 2 teams.
+
+	-- first, check if the unit is in any enemy radar
+	if (unitInEnemyRadar[unitID] == nil) then
+		return 0
+	end
+
+	if (tableLen(unitInEnemyRadar[unitID]) == 0) then
+		-- do NOT spin up a scriptDelayByFrames. When unit re-enters radar a new customWobble will be spun up
+		return 0
+	end
+
+	-- then, check if this is the "correct" DelayByFrames customWobble function to run.
+	-- Currently, there is no way to "signal-kill" a pending DelayByFrames function, so we want to make sure only one customWobble is spun up for each unit
+	if unitPendingWobble[unitID] ~= spGetGameFrame() then
+		return 0
+	end
+
+	-- Next, determine if the unit is in enemy advanced radar
+	local unitAllyTeam = unitAllyTeamCache[unitID]
+	local unitPointX, unitPointY, unitPointZ = spGetUnitPosition(unitID) -- use basepoint as radar "paints" the ground
+	local wobbleRadiusFraction = 1
+
+	for advancedRadar, tableValues in pairs(advancedRadars) do
+		local stunned_or_inbuild = spGetUnitIsStunnedCache(advancedRadar)
+		if (stunned_or_inbuild == false and unitAllyTeam ~= unitAllyTeamCache[advancedRadar]) then
+			-- check if the unit is within cylinder distance of the radar
+			local basePointX, basePointY, basePointZ, midPointX, midPointY, midPointZ = spGetUnitPositionCache(advancedRadar)
+			local dist2Dsq = (basePointX-unitPointX)^2 + (basePointZ-unitPointZ)^2
+			if ((tableValues.radarRadius+unitSpeeds[unitID])^2 >= dist2Dsq) then
+				-- +unitspeed on radarRadius to help catch units just walking into radar range, it seems unitspeed is accounted for when determining when to trigger UnitEnteredRadar
+				local raylength,_,_,_ = spTraceRayGroundBetweenPositions(basePointX, midPointY+tableValues.radarEmitHeight, basePointZ, unitPointX, unitPointY + 5, unitPointZ, false)
+				-- use midPointY to match engine, `const float losHeight  = std::max(unit->midPos.y + emitHeight, 0.0f);`
+				-- +5 on unitPointY to match engine `constexpr float LOS_BONUS_HEIGHT = 5.0f;`
+				if raylength == nil then -- returns nil if no ground collision detected
+					wobbleRadiusFraction = mMax(0.0000152587890625,mMin(wobbleRadiusFraction,tableValues.wobbleRatio))
+					-- arbitrary declare smallest possible wobbleRadiusFraction = 1/2^16 [so that a zero-vector is not created]
```

Comment:
1) whats wrong with a zero vector
2) do `local EPSILON = 1 / (2 ^ 16)` if the value being 2^n is somehow salient. otherwise just do 0.0001 instead of fake precision